### PR TITLE
Improve configuration classes

### PR DESCRIPTION
**Pull request by dejanb**
While working on Quarkus PoC I found some improvements we need to in our configuration classes. The idea is to bring that in as a separate PR (as some other improvements not related to Quarkus directly).

The main changes are:
* Defining default values for every property
* Adapting proper property names with getter/setter methods
* Using better suited classes for properties

### CODE REVIEW

File: core/src/main/java/org/eclipse/hono/config/AbstractConfig.java [hunk 1]
```diff
@@ -41,25 +41,23 @@
...
      *  A logger to be shared with subclasses.
      */
-    protected final Logger LOG = LoggerFactory.getLogger(getClass());
+    public static final Logger LOG = LoggerFactory.getLogger(AbstractConfig.class);
 
-    private String trustStorePath;
-    private char[] trustStorePassword;
-    private TrustOptions trustOptions;
+    private String trustStorePath = "";
+    private String trustStorePassword = "";
```
Comment (calohmn) [inline core/src/main/java/org/eclipse/hono/config/AbstractConfig.java:47]: I think it's common practice to rather use `char[]` for password fields.

File: core/src/main/java/org/eclipse/hono/config/AbstractConfig.java [hunk 2]
```diff
@@ -41,25 +41,23 @@
     /**
      *  A logger to be shared with subclasses.
      */
-    protected final Logger LOG = LoggerFactory.getLogger(getClass());
+    public static final Logger LOG = LoggerFactory.getLogger(AbstractConfig.class);
```
Comment (calohmn) [inline core/src/main/java/org/eclipse/hono/config/AbstractConfig.java:44]: Can you explain the motivation here? With the old definition, there is the added info in the log from which actual (sub)class the log statements are coming.

Comment (dejanb) [reply core/src/main/java/org/eclipse/hono/config/AbstractConfig.java]: That was left from the PoC phase as there were some issues with native image. Now everything works fine, so I reverted it back.

File: core/src/main/java/org/eclipse/hono/config/AbstractConfig.java (hunk 1 above)
Comment (dejanb) [reply core/src/main/java/org/eclipse/hono/config/AbstractConfig.java:47]: As all other libraries we are passing these passwords to are keeping them as Strings in memory, I don't think we're getting any benefits from keeping them as char arrays. 
And if we still chose to do so, I think getter/setter methods should follow that as well. 
Anyway, in my opinion we're just keeping a lot of transformation code that gives us no real benefits.

Comment (calohmn) [reply core/src/main/java/org/eclipse/hono/config/AbstractConfig.java:47]: Ok.

File: core/src/main/java/org/eclipse/hono/config/AuthenticatingClientConfigProperties.java [hunk 3]
```diff
@@ -29,14 +30,15 @@
...
     public static final String SERVER_ROLE_UNKNOWN = "unknown";
 
-    private String credentialsPath;
+    private String credentialsPath = "";
     private String host = "localhost";
     private boolean hostnameVerificationRequired = true;
-    private char[] password;
-    private int port;
+    private String password = "";
+    private int port = Constants.PORT_AMQPS;
```
Comment (sophokles73) [inline core/src/main/java/org/eclipse/hono/config/AuthenticatingClientConfigProperties.java:37]: what's the reason for having to use an empty string as the initial value?

Comment (dejanb) [reply core/src/main/java/org/eclipse/hono/config/AuthenticatingClientConfigProperties.java:37]: @sophokles73 At the moment with way we use Quarkus, it is complaining when properties don't have default values. So this is a workaround. I haven't yet raised this with the Quarkus devs, but I can do that and we can revert it in the future (I'm not too fond of it myself either).

Comment (sophokles73) [reply core/src/main/java/org/eclipse/hono/config/AuthenticatingClientConfigProperties.java:37]: yes, please do so. Honestly, I hate the guts out of it ;-)

File: core/src/main/java/org/eclipse/hono/config/AbstractConfig.java [hunk 4]
```diff
@@ -186,25 +175,21 @@ public final TrustOptions getTrustOptions() {
...
             LOG.debug("using certificates from PKCS12 key store [{}] as trust anchor", this.trustStorePath);
-            trustOptions = new PfxOptions()
+            return new PfxOptions()
                     .setPath(getTrustStorePath())
                     .setPassword(getTrustStorePassword());
-            break;
         case JKS:
             LOG.debug("using certificates from JKS key store [{}] as trust anchor", this.trustStorePath);
-            trustOptions = new JksOptions()
+            return new JksOptions()
```
Comment (sophokles73) [inline core/src/main/java/org/eclipse/hono/config/AbstractConfig.java:186]: I do see value in creating the options once only. However, we then should also return the already created instance in subsequent invocations of this method. WDYT?

File: core/src/main/java/org/eclipse/hono/config/KeyLoader.java [hunk 5]
```diff
@@ -162,10 +162,10 @@ public static KeyLoader fromFiles(final Vertx vertx, final String keyPath, final
         PrivateKey privateKey = null;
         List<Certificate> certChain = null;
 
-        if (keyPath != null) {
+        if (keyPath != null && !keyPath.isEmpty()) {
```
Comment (sophokles73) [inline core/src/main/java/org/eclipse/hono/config/KeyLoader.java:165]: StringUtils.isNullOrEmpty()?

File: core/src/main/java/org/eclipse/hono/config/ProtocolAdapterProperties.java [hunk 6]
```diff
@@ -36,7 +36,7 @@
     private long eventLoopBlockedCheckTimeout = 5000L;
     private int maxConnections = 0;
     private Duration tenantIdleTimeout = DEFAULT_TENANT_IDLE_TIMEOUT;
-    private Map<String, MapperEndpoint> mapperEndpoints = new HashMap<>();
+    private HashMap<String, MapperEndpoint> mapperEndpoints = new HashMap<>();
```
Comment (sophokles73) [inline core/src/main/java/org/eclipse/hono/config/ProtocolAdapterProperties.java:39]: We usually use an abstract type for field declarations, don't we? Why can't we keep the `Map` here?

File: core/src/main/java/org/eclipse/hono/config/AuthenticatingClientConfigProperties.java (hunk 3 above)
Comment (dejanb) [reply core/src/main/java/org/eclipse/hono/config/AuthenticatingClientConfigProperties.java]: Setting them explicitly to null also does the trick and it looks a bit nicer. So let's go with that for the moment and push for the fix.

File: core/src/main/java/org/eclipse/hono/config/ProtocolAdapterProperties.java (hunk 6 above)
Comment (dejanb) [reply core/src/main/java/org/eclipse/hono/config/ProtocolAdapterProperties.java:39]: We're basically hitting this with using the Map: https://github.com/quarkusio/quarkus/issues/8017. And that's not considered as the issue at the moment. We can create our own converter perhaps and I'll ask if there are a better way to do all this, but to me this is a quick/easy workaround that we can use for now.

**Pull request merged**